homeassistant/components/homekit/type_media_players.py

```diff
@@ -0,0 +1,146 @@
+"""Class to hold all media player accessories."""
+import logging
+
+from pyhap.const import CATEGORY_SWITCH
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID, CONF_MODE, SERVICE_MEDIA_PAUSE, SERVICE_MEDIA_PLAY,
+    SERVICE_MEDIA_STOP, SERVICE_TURN_OFF, SERVICE_TURN_ON, SERVICE_VOLUME_MUTE,
+    STATE_OFF, STATE_PLAYING, STATE_UNKNOWN)
+from homeassistant.components.media_player import (
+    ATTR_MEDIA_VOLUME_MUTED, DOMAIN)
+
+from . import TYPES
+from .accessories import HomeAccessory
+from .const import (
+    CHAR_NAME, CHAR_ON, ON_OFF, PLAY_PAUSE, PLAY_STOP, SERV_SWITCH,
+    TOGGLE_MUTE)
+
+_LOGGER = logging.getLogger(__name__)
+
+MODE_FRIENDLY_NAME = {ON_OFF: 'Power',
+                      PLAY_PAUSE: 'Play/Pause',
+                      PLAY_STOP: 'Play/Stop',
+                      TOGGLE_MUTE: 'Mute'}
+
+
+@TYPES.register('MediaPlayer')
+class MediaPlayer(HomeAccessory):
+    """Generate a Media Player accessory."""
+
+    def __init__(self, *args):
+        """Initialize a Switch accessory object."""
+        super().__init__(*args, category=CATEGORY_SWITCH)
+        self._flag = {ON_OFF: False, PLAY_PAUSE: False,
+                      PLAY_STOP: False, TOGGLE_MUTE: False}
+
+        self.chars = {ON_OFF: None, PLAY_PAUSE: None,
+                      PLAY_STOP: None, TOGGLE_MUTE: None}
+
+        modes = self.config.get(CONF_MODE)
```

Comment:
Through the change `CONF_MODE` will always be set `modes = self.config[CONF_MODE]`. (The empty line above can be deleted as well.)